static/src/javascripts/projects/common/modules/commercial/adfree/maintain-adfree-freshness.js

```diff
@@ -0,0 +1,36 @@
+define([
+    'common/utils/cookies',
+    'common/utils/config',
+    'common/utils/storage',
+    'common/modules/identity/api',
+    'common/modules/commercial/adfree/renew-adfree-status'
+], function (
+    cookies,
+    config,
+    storage,
+    identity,
+    renewAdfreeStatus
+) {
+    return maintainAdfreeFreshness;
+
+    function maintainAdfreeFreshness() {
+        if (featureEnabled() && identity.isUserLoggedIn() && userNeedsNewAdfreeCookie()) {
+            renewAdfreeStatus.renew();
+        }
+    }
+
+    function userNeedsNewAdfreeCookie() {
+        var adfreeCookie = cookies.get('gu_adfree_user');
+        return (adfreeCookie === null) || adfreeCookieStale();
+    }
+
+    function adfreeCookieStale() {
+        var adfreeCookieExpiry = storage.local.get('gu.adfree.user.expiry'),
```

Comment:
@jbreckmckye I was just chatting with @ostapneko and we were wondering why we were setting the cookie in the first place. I'd also prefer to see this all stored in local storage so it sounds like we're all of the same opinion. Worth having a quick chat this afternoon?